### PR TITLE
[wasm] Bitcode code generation improvements

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by vargaz** [comment]
@monojenkins build failed

File: mono/mini/mini-llvm.c [hunk 1]
```diff
@@ -2477,17 +2507,49 @@ emit_get_method (MonoLLVMModule *module)
...
 	LLVMBuilderRef builder = LLVMCreateBuilder ();
+	LLVMValueRef table = NULL;
 	char *name;
 	int i;
+	gboolean emit_table = FALSE;
+
+#ifdef TARGET_WASM
+	/*
+	 * Emit a table of functions instead of a switch statement,
+	 * its very efficient on wasm. This might be usable on
```
Comment (jeromelaban) [inline mono/mini/mini-llvm.c]: In which way is this more efficient ?

File: mono/mini/aot-runtime.c [hunk 2]
```diff
@@ -4548,9 +4546,10 @@ static void
 init_llvmonly_method (MonoAotModule *amodule, guint32 method_index, MonoMethod *method, MonoClass *init_class, MonoGenericContext *context)
 {
 	ERROR_DECL (error);
+	gboolean res;
 
-	init_method (amodule, method_index, method, init_class, context, error);
-	if (!is_ok (error)) {
+	res = init_method (amodule, method_index, method, init_class, context, error);
```
Comment (luhenry) [inline mono/mini/aot-runtime.c]: Shouldn't we guarantee that if `res != TRUE`, then `error` is always set to something not-ok?

**Comment by luhenry** [comment]
@monojenkins build failed

File: mono/mini/mini-llvm.c (hunk 1 above)
Comment (vargaz) [reply mono/mini/mini-llvm.c]: On wasm, each function pointer is an int and it doesn't require runtime linking, so the whole thing is going to be an array of ints.

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**